python/lsst/obs/pfs/isrTask.py

```diff
@@ -442,6 +448,76 @@ def run(self, ccdExposure, **kwargs):
 
         return results
 
+    def runH4RG(self, exposure, defects=None, **kwargs):
+        """Specialist instrument signature removal for H4RG detectors
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The raw exposure that is to be run through ISR.  The
+            exposure is modified by this method.
+        defects : `lsst.ip.isr.Defects`, optional
+            List of defects.
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with component:
+            - ``exposure`` : `afw.image.Exposure`
+                The fully ISR corrected exposure.
+            - ``outputExposure`` : `afw.image.Exposure`
+                An alias for `exposure`
+        """
+        for k, v in kwargs.items():
+            if v is not None:
+                if k == "fringes" and v.fringes is None:
+                    continue
+
+                if k not in ["camera", ]:
+                    self.log.warn("Unexpected argument for runH4RG: %s", k)
+
+        if self.config.doDefect and defects is None:
+            raise RuntimeError("Must supply defects if config.doDefect=True.")
+        if self.config.doIPC and defects is None:
+            raise RuntimeError("Must supply defects if config.doIPC=True.")
+
+        if self.config.doIPC:
+            self.correctIPC(exposure, defects, self.config.ipcCoeffs)
+
+        if self.config.doDefect:
+            super().maskAndInterpolateDefects(exposure, defects)
+
+        return pipeBase.Struct(exposure=exposure,
+                               outputExposure=exposure,  # is this needed? Cargo culted from ip_isr isrTask.py
+                               flattenedThumb=None,
+                               ossThumb=None,
+                               )
+
+    def correctIPC(self, exposure, defects, ipcCoeffs):
+        ipc_cx, ipc_cy = ipcCoeffs
+
+        ipc = exposure.maskedImage.clone()
```

Comment:
Hmm.   I'd have to think about the proper way to handle the variance.  It is affected by IPC, but the current code (as Paul implies) doesn't deal with it.  Probably better to leave the clone here, and fix the values later.  PIPE2D-1091

Reply:
I was just saying that copying the variance is unnecessary and leaving it out might give a mild speed boost.

Reply:
Well, it's only unnecessary because we aren't properly setting it, yet.  So I think we should eat the small speed/memory penalty in exchange for more correct code, ready for PIPE2D-1091